code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -178,4 +178,61 @@
 	green = max(min(green + rand (-25, 25), 255), 0)
 	blue = max(min(blue + rand (-25, 25), 255), 0)
 
-	return list(red, green, blue)
\ No newline at end of file
+	return list(red, green, blue)
+
+/mob/living/carbon/human/proc/pick_gender(var/mob/user)
+	var/new_gender = alert(src, "Please select gender.", "Character Generation", "Male", "Female")
+	if (new_gender)
+		setGender(new_gender == "Male" ? MALE : FEMALE)
+	update_body()
+
+/mob/living/carbon/human/proc/pick_appearance(var/mob/user)
+	var/new_tone = input(src, "Please select skin tone level: 1-220 (1=albino, 35=caucasian, 150=black, 220='very' black)", "Character Generation")  as text
+	if (!new_tone)
+		new_tone = 35
+	my_appearance.s_tone = max(min(round(text2num(new_tone)), 220), 1)
+	my_appearance.s_tone =  -src.my_appearance.s_tone + 35
+
+	var/new_facial = input(user, "Please select facial hair color.", "Character Generation") as color
+	if(new_facial)
+		my_appearance.r_facial = hex2num(copytext(new_facial, 2, 4))
+		my_appearance.g_facial = hex2num(copytext(new_facial, 4, 6))
+		my_appearance.b_facial = hex2num(copytext(new_facial, 6, 8))
+
+	var/new_hair = input(user, "Please select hair color.", "Character Generation") as color
+	if(new_facial)
+		my_appearance.r_hair = hex2num(copytext(new_hair, 2, 4))
+		my_appearance.g_hair = hex2num(copytext(new_hair, 4, 6))
+		my_appearance.b_hair = hex2num(copytext(new_hair, 6, 8))
+
+	var/new_eyes = input(user, "Please select eye color.", "Character Generation") as color
+	if(new_eyes)
+		my_appearance.r_eyes = hex2num(copytext(new_eyes, 2, 4))
+		my_appearance.g_eyes = hex2num(copytext(new_eyes, 4, 6))
+		my_appearance.b_eyes = hex2num(copytext(new_eyes, 6, 8))
+
+	// hair
+	var/list/all_hairs = typesof(/datum/sprite_accessory/hair) - /datum/sprite_accessory/hair
+	var/list/hairs = list()
```

Comment:
is this used for anything? It seems this list is being populated but never used

Reply:
good catch!